### PR TITLE
Include all parameter paths in `toDataFrame()` for grouped PIParameters

### LINKED ISSUE #230: Missing parameter path in results data frame after grouping
When grouping different parameters from the same simulation into one PIParameter, only the first parameter path is present in the dataframe after calling piResults$toDataFrame(). 
This seems not to be an issue with the PI itself, but rather with the creation of the dataframe. 

In a minimal example (code below), after running the piTask one sees that grouped parameters are both updated to the same value (Note: original values were 1 and 1):

Image

However, printing the piResults$toDataFrame() yields:

Image

Thus, listing only the first parameter path of the PIParameter.

Minimal example:

```{r}
library(ospsuite)
library(ospsuite.parameteridentification)

# load sim
sim_250mg <- loadSimulation(system.file("extdata", "Aciclovir.pkml", package = "ospsuite"))

# Path to the dose parameter
doseParameterPath <- "Applications|IV 250mg 10min|Application_1|ProtocolSchemaItem|Dose"
# Get the parameter from the first simulation

# Get the instances of the parameters
sim_250mg_doseParam <- getParameter(path = doseParameterPath, container = sim_250mg)
print(sim_250mg_doseParam)


# Creating a PIParameters object with two different simulation parameters retrieved from the same simulations
piParameterPermeabilityFactors <- PIParameters$new(parameters = list(
  getParameter(path = "Aciclovir|P_bc_pls_factor", container = sim_250mg),
  getParameter(path = "Aciclovir|P_pls_bc_factor", container = sim_250mg)
))
print(piParameterPermeabilityFactors)

# load data
filePath <- system.file("extdata", "Aciclovir_Profiles.xlsx", package = "ospsuite.parameteridentification")

# Create importer configuration for the file
importConfig <- createImporterConfigurationForFile(filePath = filePath)
# Set naming patter
importConfig$namingPattern <- "{Source}.{Sheet}.{Dose}"
# Import data sets
obsData <- loadDataSetsFromExcel(xlsFilePath = filePath, importerConfigurationOrPath = importConfig, importAllSheets = TRUE)

# mapping
simOutputPath <- "Organism|PeripheralVenousBlood|Aciclovir|Plasma (Peripheral Venous Blood)"
outputMapping_250mg <- PIOutputMapping$new(quantity = getQuantity(path = simOutputPath, container = sim_250mg))
outputMapping_250mg$addObservedDataSets(obsData$`Aciclovir_Profiles.Vergin 1995.Iv.250 mg`)
outputMapping_250mg$scaling <- "log"

# pi config
piConfiguration <- PIConfiguration$new()

# setup task
piTask <- ParameterIdentification$new(
  simulations = list(sim_250mg),
  parameters = list(piParameterPermeabilityFactors),
  outputMappings = list(outputMapping_250mg),
  configuration = piConfiguration
)
print(piTask$parameters[[1]]$parameters)

# run PI
piResult <- piTask$run()
print(piTask$parameters[[1]]$parameters)
print(piResult$toDataFrame())

```

### PR DESCRIPTION
**Pull request by rengelke**
Closes #230

## Summary by CodeRabbit

* **Bug Fixes**
  * Dataframe export now emits one row per parameter-path entry when parameters are grouped; clearer error is raised if parameter metadata is missing.

* **Tests**
  * Added tests covering grouped-parameter dataframe expansion and the missing-parameter-metadata error.

* **Documentation**
  * Changelog updated to document the modified dataframe row-expansion behavior.

* **Style**
  * Minor code formatting adjustments.

### CODE REVIEW

File: tests/testthat/test-pi-result.R [hunk 1]
```diff
@@ -33,3 +48,28 @@ test_that("print() produces expected output", {
...
+  )
+
+  piResult <- PIResult$new(
+    optimResult = optimResult,
+    piParameters = list(piParameterLipo)
+  )
+
+  df <- piResult$toDataFrame()
+
+  expect_equal(nrow(df), 2)
```
Comment (nautilus69) [inline tests/testthat/test-pi-result.R:72]: Not sure if this can be an edge case or not: 
expect_equal(length(unique(df$path)), 2)

or instead, Checking the actual path strings. Without it, the test would pass even if both rows had the same path (which wouldn't happen currently, but makes the test's intent more explicit).

**Review by nautilus69 — APPROVED** [review]
Approved.

File: tests/testthat/test-pi-result.R (hunk 1 above)
Comment (rengelke) [reply tests/testthat/test-pi-result.R:72]: The existing assertion is sufficient. It directly proves that deduplication no longer drops the second row, which is        exactly what the bug was about.

**Comment by PavelBal** [comment]
@rengelke all checks are performed twice?
